components/collector/tests/source_collectors/quality_time/test_missing_metrics.py

```diff
@@ -63,35 +83,36 @@ def subject_metrics(self, subject_type) -> list[str]:
 
     def create_entity(self, report, subject_uuid: str, subject, metric_type: str) -> dict[str, str]:
         """Create a missing metric entity."""
+        data_model_subject = self.subject_type(self.data_model["subjects"], subject["type"]) or {"name": "missing"}
         return {
             "key": f"{report['report_uuid']}:{subject_uuid}:{metric_type}",
             "report": report["title"],
             "report_url": f"https://quality_time/{report['report_uuid']}",
             "subject": subject["name"],
             "subject_url": f"https://quality_time/{report['report_uuid']}#{subject_uuid}",
             "subject_uuid": f"{subject_uuid}",
-            "subject_type": self.data_model["subjects"][subject["type"]]["name"],
+            "subject_type": data_model_subject["name"],
             "metric_type": self.data_model["metrics"][metric_type]["name"],
         }
 
+    def subject_type(self, subjects: dict, subject_type_name: str):
+        """Return the data model subject with the specified type."""
+        for name, subject_type in subjects.items():
+            if name == subject_type_name:
+                return subject_type
+            if composite_subject_type := self.subject_type(subject_type.get("subjects", {}), subject_type_name):
```

Comment:
Isn't this re-implementing the code it's trying to test?

Reply:
Yeah, the alternative is to hardcode the expected numbers, but since they are pretty large it's hard to see they are correct. Not sure how to resolve this, to be honest.

Reply:
I've removed at least a bit of the duplication.